packages/root-cms/ui/hooks/useStuckObserver.ts

```diff
@@ -0,0 +1,35 @@
+import {useEffect, useRef, useState} from 'preact/hooks';
+
+interface StuckObserverOptions {
+  offsetHeight: () => number;
+}
+
+/**
+ * Provides an observer that detects when an element is "stuck" at the top of the viewport.
+ */
+export function useStuckObserver(options: StuckObserverOptions) {
```

Comment:
Since this is for observing `position: sticky` elements I think I would call this `useStickyObserver()`